app/src/main/java/com/nervousfish/nervousfish/activities/SettingsActivity.java

```diff
@@ -0,0 +1,264 @@
+package com.nervousfish.nervousfish.activities;
+
+
+import android.annotation.TargetApi;
+import android.content.Intent;
+import android.os.Build;
+import android.os.Bundle;
+import android.preference.ListPreference;
+import android.preference.Preference;
+import android.preference.PreferenceActivity;
+import android.support.v7.app.ActionBar;
+import android.preference.PreferenceFragment;
+import android.preference.PreferenceManager;
+import android.view.MenuItem;
+
+import com.nervousfish.nervousfish.ConstantKeywords;
+import com.nervousfish.nervousfish.R;
+import com.nervousfish.nervousfish.data_objects.Profile;
+import com.nervousfish.nervousfish.service_locator.IServiceLocator;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.util.List;
+
+/**
+ * A {@link PreferenceActivity} that presents a set of application settings. This
+ * is the place where your profile is changed.
+ */
+@SuppressWarnings({"checkstyle:AnonInnerLength", "PMD.AvoidUsingVolatile"})
+//1. In this class large anonymous classes are needed. It does not infer with readability.
+//2. The volatile identifier is needed because this class uses static methods, which are essential.
+public final class SettingsActivity extends AAppCompatPreferenceActivity {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger("SettingsActivity");
+    private static boolean firstLoad = true;
+    private static final String DISPLAY_NAME = "display_name";
+    private static volatile IServiceLocator serviceLocator;
+
+    /**
+     * A preference value change listener that updates the preference's summary
+     * to reflect its new value.
+     */
+    private static Preference.OnPreferenceChangeListener sBindPreferenceSummaryToValueListener = new Preference.OnPreferenceChangeListener() {
+        @Override
+        public boolean onPreferenceChange(final Preference preference, final Object value) {
+            LOGGER.info("Preference changed");
+            final String stringValue = value.toString();
+
+            if (preference.getKey().equals(DISPLAY_NAME)) {
+                LOGGER.info("Preference changed at the display name");
+                updateDisplayName(preference, stringValue);
+            } else if (preference instanceof ListPreference) {
+                LOGGER.info("Preference changed for a ListPreference");
+                // For list preferences, look up the correct display value in
```

Comment:
Can you abstract this into its own implementation similar to [when the display name is changed](https://github.com/ericcornelissen/NervousFish/pull/214/files#diff-bf06f4c02cc485ea11575068ccada50eR53)